app/clients.go

```diff
@@ -0,0 +1,128 @@
+package app
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/OAuth2withJWT/identity-provider/app/validation"
+)
+
+type ClientService struct {
+	repository ClientRepository
+}
+
+type Client struct {
+	Id          string
+	Name        string
+	Secret      string
+	Scope       string
+	RedirectURI string
+	CreatedBy   int
+}
+
+type CreateClientRequest struct {
+	Name        string
+	Scope       string
+	RedirectURI string
+	CreatedBy   int
+}
+
+type ClientCredentials struct {
+	Id     string
+	Secret string
+}
+
+func (req *CreateClientRequest) validateClientRegistrationFields(s *ClientService) error {
```

Comment:
Why not just name the function `validate`?  Then it would read `req.validate(...)` instead of `req.validateClientRegistrationFields`

Reply:
I tried to follow the same naming as in `Users.go`.